pkg/node/manager.go

```diff
@@ -32,29 +37,64 @@ func GetNode(ni Identity) *Node {
 	return n
 }
 
+func deleteNodeCIDR(ip *net.IPNet) {
+	if ip == nil {
+		return
+	}
+
+	if err := tunnel.DeleteTunnelEndpoint(ip.IP); err != nil {
+		log.Debugf("bpf: Unable to delete %s in tunnel endpoint map: %s", ip, err)
+	}
+}
+
+func updateNodeCIDR(ip *net.IPNet, host net.IP) {
+	if ip == nil {
+		return
+	}
+
+	if err := tunnel.SetTunnelEndpoint(ip.IP, host); err != nil {
+		log.Debugf("bpf: Unable to update %s in tunnel endpoint map: %s", ip, err)
+	}
+}
+
 // UpdateNode updates the new node in the nodes' map with the given identity.
 func UpdateNode(ni Identity, n *Node) {
 	mutex.Lock()
-	//if oldNode, ok := nodes[ni]; ok {
-	// oldNode
-	// remove oldNode metadata (IPs-> endpoints CIDR) from bpf map
-	//}
+	if oldNode, ok := nodes[ni]; ok {
+		deleteNodeCIDR(oldNode.IPv4AllocCIDR)
+		deleteNodeCIDR(oldNode.IPv6AllocCIDR)
+	}
 
 	// FIXME if PodCIDR is empty retrieve the CIDR from the KVStore
 
-	// add new node changes to bpf map.
-	// **note**: this will eventually add pod-cidr routes for the own node.
+	log.Debugf("bpf: Setting tunnel endpoint %+v: %+v %+v",
+		n.GetNodeIP(false), n.IPv4AllocCIDR, n.IPv6AllocCIDR)
+
+	nodeIP := n.GetNodeIP(false)
+	updateNodeCIDR(n.IPv4AllocCIDR, nodeIP)
+	updateNodeCIDR(n.IPv6AllocCIDR, nodeIP)
 
 	nodes[ni] = n
-	mutex.Unlock()
 
+	mutex.Unlock()
 }
 
 // DeleteNode remove the node from the nodes' maps.
 func DeleteNode(ni Identity) {
-	// remove node from bpf map
-
 	mutex.Lock()
+	if n, ok := nodes[ni]; ok {
+		log.Debugf("bpf: Removing tunnel endpoint %+v: %+v %+v",
+			n.GetNodeIP(false), n.IPv4AllocCIDR, n.IPv6AllocCIDR)
+
+		if n.IPv4AllocCIDR != nil {
+			tunnel.DeleteTunnelEndpoint(n.IPv4AllocCIDR.IP)
+		}
+
+		if n.IPv6AllocCIDR != nil {
+			tunnel.DeleteTunnelEndpoint(n.IPv6AllocCIDR.IP)
+		}
+	}
+
 	delete(nodes, ni)
```

Comment:
Agreed. I think this should also use the state mechanism once we have it which then attempts to delete multiple times. I will change this to keep the node for now.